CodeGen/Program.cs

```diff
@@ -77,7 +77,19 @@ public static int Main(bool verbose = false, DirectoryInfo? repositoryRoot = nul
                 if (!skipNanoFramework)
                 {
                     Log.Information("Generate nanoFramework projects\n---");
-                    NanoFrameworkGenerator.Generate(rootDir, quantities, updateNanoFrameworkDependencies);
+                    NanoFrameworkGenerator.Generate(rootDir, quantities);
+
+                    // after generating code, force a dependencies update to make sure it's using the latest NuGet packages
+                    updateNanoFrameworkDependencies = true;
```

Comment:
Not sure about this line. The default behavior when running `generate-code.sh` is to also generate code for nanoFramework. If I understand it right, it will then also bump the version numbers for nanoFramework in the codegen if newer versions are available.

When someone creates a PR on a new quantity or unit, this will just cause confusion as this change is entirely unrelated to the PR.

Reply:
Also, while this works, it seems a bit meta for the codegen to update its own source code.
Can't we look up the current versions from .nfproj or .nuspec files instead of having it stored in codegen source code?

Reply:
Yes, when adding a new quantity it will run the update references as part of the process. 
This intent was to have it automatically checked and updated to be sure it was using the latest available version.

As for the auto-update, yes again, it was an intended to have the class updating itself to always have the latest version available when the generated code step was run.

And yes, as it is now it grabs the "current" base version strings from the nfproj in order to fill in the nuspec and packages files.
When pulling the NuGet versions from nuget.org we only have access to the package version. The assembly version (that's used on the nfproj file here: _<Reference Include="mscorlib, Version=**1.10.5.4** ..._  can't be pulled from anywhere except from the DLL file itself. The previous update strategy had this flaw, that's why it ended up failing like it did. Because the automated update was only updating the nuget version. Leaving it up to the human doing it the job of updating the version (which wasn't being done).

But nuget CLI is smart enough to handle that and it does update both versions: assembly and nuget. It's therefore a superior way of dealing with the automated updates. 

If you prefer, I can remove this part of the self update of the base versions. Or we can have it not run automatically when generating code. 
Let me know. 😄

Reply:
@josesimoes 
Thanks for clarifying. I may not have been clear in what I was trying to point out.

1. You are now hard coding `internal static string MscorlibVersion = "1.10.5.0";` etc. instead of looking it up from `Angle.nfproj`.
2. Codegen is run, using hardcoded values. Project files, nuspec files, packages.config files, etc.
3. The new version is determined by `nuget.exe update` instead of nuget.org APIs.
4. The new version is stored in `NanoFrameworkGenerator.cs`, replacing the hard coded value. (Presumably used on next codegen).
5. Nuspec files are written (again) looking up versions from `packages.config` after the `nuget.exe update`.

I believe step 1 & 4 could be dropped. 

I pushed a suggestion to this in my branch: d610dfe4b


Just to comment on the `nuget update` approach vs nuget API. Overall, I do like that it is simpler. One caveat is that it is much slower and takes about 30-40 seconds on my machine. It seems to spend the time updating N projects in sequence. As long as this does not run on every codegen, I think it's fine. Another option is to go the nuget API route and download the nuget and inspect the .dll to find the assembly version as you described. I'm fine with either way.

Reply:
Glad that we're getting there! 

> 1. You are now hard coding `internal static string MscorlibVersion = "1.10.5.0";` etc. instead of looking it up from `Angle.nfproj`. ✅ 

> 2. Codegen is run, using hardcoded values. Project files, nuspec files, packages.config files, etc.
✅  

> 3. The new version is determined by `nuget.exe update` instead of nuget.org APIs. ✅ 

> 4. The new version is stored in `NanoFrameworkGenerator.cs`, replacing the hard coded value. (Presumably used on next codegen). ✅ 

> 5. Project and nuspec files are written to (again) by looking up versions from `packages.config` after the `nuget.exe update`. ✅ 

 
> I believe step 1 & 4 could be dropped.

By dropping this "memory" field we are left with parse one of the existing nfproj files to grab the "last" version and use that to generate the various files for the current iteration. It seems a bit more inefficient, but I get it. You don't want the code changing every time the version changes. 

> I pushed a suggestion to this in my branch: [d610dfe](https://github.com/angularsen/UnitsNet/commit/d610dfe4b03b9a114e00bcc8cc5ee6f7f9c1b7cf)

Got you! Looks good to me. Let's go with it. 


> Just to comment on the `nuget update` approach vs nuget API. Overall, I do like that it is simpler. One caveat is that it is much slower and takes about 30-40 seconds on my machine. It seems to spend the time updating N projects in sequence. As long as this does not run on every codegen, I think it's fine. Another option is to go the nuget API route and download the nuget and inspect the .dll to find the assembly version as you described. I'm fine with either way.

As it is, it will take the "long run" (to perform the nuget CLI run) only if the `updateNanoFrameworkDependencies` is passed. Which I believe it's OK.

And despite the longest execution time using nuget CLI I believe it's easier and requiring less code maintenance, when compared with the approach of having to call the nuget API, download the package, extract it, peek the version from the dll file! 😅

Reply:
Perfect, I think we agree then 🤝 😄